docs/content/direct/user-guide.md

```diff
@@ -31,3 +31,59 @@ Additionally, the recipe for registering a WEC with the ITS needs to be modified
 ## When everythig is not on the same machine
 
 Thus far we can only say how to handle this when the hosting cluster is OpenShift. The problem is getting URLs that work from everywhere. OpenShift is a hosted product, your clusters have domain names that are resolvable from everywhere. In other words, if you use an OpenShift cluster as your hosting cluster then this problem is already solved.
+
+## Rule-based customization
+
+KubeStellar can distribute one workload object to multiple WECs, and it is not uncommon for users to need some customization to each WEC. By _rule based_ we mean that the customization is not expressed via one or more literal expressions but rather can refer to _properties_ of each WEC by property name. As KubeStellar distributes or transports a workload object from WDS to a WEC, the object can be transformed in a way that depends on those properties.
+
+KubeStellar currently has a simple but limited way to specify rule-based customization, called "template expansion". This works with templates as defined in the "text/template" standard package of Go, and is applied to each leaf string of an object independently. More expressive customization will be developed in the future.
+
+### Template Expansion
+
+Template expansion is an optional feature that a user can request on an object-by-object basis. The way to request this feature on an object is to put the following annotation on the object.
+
+```yaml
+    control.kubestellar.io/expand-templates: "true"
+```
+
+The customization that template expansion does when distributing an object from a WDS to a WEC is applied independently to each leaf string of the object. The string is first parsed as a template, and then replaced with the result of expanding the template (using the `Execute` method of a `*Template`) using a `map` holding key-value pairs conveying properties of the WEC. For the purposes of template expansion, the "properties" of a WEC are defined by the labels and annotations of the inventory object (`ManagedCluster`) representing that WEC in the inventory. Labels take precedence over annotations. Errors from this process are reported in the status field of the Binding object involved.
```

Comment:
this file is called `user-guide` but includes way too much implementation details.
a user-guide typically describes for a user what is the feature and how to use it. 
her the description includes which function of which package is used internally. 
```
using the `Execute` method of a `*Template`
```
a user-guide usually doesn't include implementation details. I would maybe mention that the the customization is based on the Go templating `text/template` module and that's it.

Reply:
I reworded to simplify.